torch/ao/quantization/fx/_model_report/model_report_observer.py

```diff
@@ -10,6 +10,13 @@ class ModelReportObserver(ObserverBase):
     Dynamic or Static Quantization is more appropriate for their model given the general
     distributions of their data.
 
+    Args:
```

Comment:
should we have different observers for different tasks?

Reply:
That was a design decision @HDCharles and I considered, but we thought it would make more sense to keep all the functionality in one observer because:
1.) it would be easier to only deal with the adding and removal of observers if we were only looking for one
2.) if the user wanted to perform a further deep dive into the extra collected statistics, it made more sense for them to only have to look for one type of observer
3.) some of the collected statistics by the observer are used multiple times in each of the observer "data collection flows" therefore it also helps cut down on redundant code

Reply:
I also think it's a bit weird for observers associated with a completely unrelated detector to have a `comp_percentile` argument when this is only used to detect outliers. Maybe we should discuss this more offline after this PR is merged.

Reply:
Separating the observers per task would also allow you to structure the code more cleanly by breaking down this giant class (right now we have to keep adding fields/methods to it every time we have a new detector)